core/imt/src/RTaskArena.cxx

```diff
@@ -127,7 +142,7 @@ ROOT::ROpaqueTaskArena &RTaskArenaWrapper::Access()
    return *fTBBArena;
 }
 
-std::shared_ptr<ROOT::Internal::RTaskArenaWrapper> GetGlobalTaskArena(unsigned maxConcurrency)
+std::shared_ptr<ROOT::Internal::RTaskArenaWrapper> GetGlobalTaskArena(bool semantic, unsigned maxConcurrency, ROOT::EIMTConfig config)
```

Comment:
I find the additional `bool semantic` confusing. Isn't this already implicit with `config = kNumConfigs` (which is horrible name for that)?

Reply:
> I find the additional bool semantic confusing.

The idea is to not really on any special value.  If we really on the value of the enum to steer this function then it is harder (or need to be moved one level up) to detect when the user passed the 'wrong' value' to the enum overload.
(this routine is meant to *only* be called internally)

>  config = kNumConfigs

We use this 'kNum[berOf]...' pattern is a few other places.   I don't have a strong preference.  What alternative spelling are you thinking of?

Reply:
But we already have this problem, the user can pass in `kNumConfigs` now...

Reply:
What prevents the user from calling `ROOT::EnableImplicitMT(ROOT::EIMTConfig::kNumConfigs)`? It's a valid enum value

Reply:
Then I would just remove the `semantic` argument, it seems redundant.